### PR TITLE
adds tail sweep to queen

### PR DESCRIPTION
**Pull request by coldironwarrior**
title

## About The Pull Request

adds tail sweep to queen, edited to give a .1 second stun instead of .5 seconds like defender
moves tail sweep to general abilities.dm

please please please code review, local keeps crashing before i can even start server

## Why It's Good For The Game

the kurtbal requested this; we were floating ideas how to make queen more fun to play since she's kinda underwhelming in combat, according to ppl

## Changelog
:cl:
add: queen has a tail sweep ability now
balance: gives queen tail sweep
/:cl:

### CODE REVIEW

**Comment by Arturlang** [comment]
Why not simply allow royal bulwark to be used while moving?

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 1]
```diff
@@ -521,3 +521,10 @@
 	scale = 0.8
 	friction = generator(GEN_NUM, 0.1, 0.15)
 	spin = generator(GEN_NUM, -20, 20)
+
+// ***************************************
+// *********** Queen tail sweep
+// ***************************************
+/datum/action/ability/xeno_action/tail_sweep/queen
+	var/swipe_stun_duration = 0.1 SECONDS
```
Comment (Xander3359) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:529]: This is not what I told you to do at all

File: code/modules/mob/living/carbon/xenomorph/castes/defender/abilities_defender.dm [hunk 2]
```diff
@@ -1,86 +1,8 @@
...
-
-		to_chat(H, span_xenowarning("We are struck by \the [X]'s tail sweep!"))
-		playsound(H,'sound/weapons/alien_claw_block.ogg', 50, 1)
-
-	addtimer(CALLBACK(X, TYPE_PROC_REF(/atom, remove_filter), "defender_tail_sweep"), 0.5 SECONDS) //Remove cool SFX
-	succeed_activate()
-	if(X.crest_defense)
-		X.use_plasma(ability_cost)
-	add_cooldown()
+/datum/action/ability/xeno_action/tail_sweep/defender
```
Comment (Xander3359) [inline code/modules/mob/living/carbon/xenomorph/castes/defender/abilities_defender.dm:4]: New datum that does... nothing....

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 3]
```diff
@@ -1493,3 +1491,87 @@
...
+	. = ..()
+	var/mob/living/carbon/xenomorph/X = owner
+	if(X.crest_defense && X.plasma_stored < (ability_cost * 2))
+		to_chat(X, span_xenowarning("We don't have enough plasma, we need [(ability_cost * 2) - X.plasma_stored] more plasma!"))
+		return FALSE
+
+/datum/action/ability/xeno_action/tail_sweep/action_activate()
+	var/mob/living/carbon/xenomorph/X = owner
+
+	GLOB.round_statistics.defender_tail_sweeps++
```
Comment (Xander3359) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1520]: I told you to change this to use a proc that you can override.

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 4]
```diff
@@ -1493,3 +1491,87 @@
...
+		H.add_filter("defender_tail_sweep", 2, gauss_blur_filter(1)) //Add cool SFX; motion blur
+		addtimer(CALLBACK(H, TYPE_PROC_REF(/atom, remove_filter), "defender_tail_sweep"), 0.5 SECONDS) //Remove cool SFX
+		var/damage = X.xeno_caste.melee_damage
+		var/affecting = H.get_limb(ran_zone(null, 0))
+		if(!affecting) //Still nothing??
+			affecting = H.get_limb("chest") //Gotta have a torso?!
+		H.knockback(X, sweep_range, 4)
+		H.apply_damage(damage, BRUTE, affecting, MELEE)
+		H.apply_damage(damage, STAMINA, updating_health = TRUE)
+		H.Paralyze(0.5 SECONDS) //trip and go
```
Comment (Xander3359) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1545]: This is where you want the var to be so you can change the amount as you want (assuming you are preserving the rest of the ability)

**Review by Xander3359 — CHANGES_REQUESTED** [review]

**Comment by Naaanii** [comment]
If she's underwhelming in combat would probably be better to go back to Ovi Queen and have her buff silos or something instead of make the role a harder to punish all round combat caste. This just kind of further creeps a caste that's gotten numerous buffs already and is quite fine in it's current state.

Adding cqc abilities or buffing numbers for the sake of creep will not suddenly make a caste more appealing if the flaw is in the design in itself.

**Comment by SimplySmiley123** [comment]
> If this gets merged but not queen slap , then TGMC has truly fallen. (both should get merged queen should be the baddest xenomorphs of the hive and should have more stuns to show that , you shouldn't ever 1 v 1 a queen, not to mention its hardcapped to 1 player and a necesarry role which is almost never taken due to how many combat abilities it lacks in general , that won't compensate for the slow speed, and the slow armor regen.

Does everything need a stun? Really?
Stuns would make it more boring if anything, more trivial.

**Comment by Barnet2** [comment]
This will be funny

File: code/modules/mob/living/carbon/xenomorph/castes/defender/abilities_defender.dm (hunk 2 above)
Comment (coldironwarrior) [reply code/modules/mob/living/carbon/xenomorph/castes/defender/abilities_defender.dm:4]: point taken but i'm picky about being specific and wanted to set it up so if anyone edited defender specifically in the future, they didn't have to edit the generic

File: code/modules/mob/living/carbon/xenomorph/abilities.dm (hunk 4 above)
Comment (coldironwarrior) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:1545]: i swore i changed it but maybe i re-pasted original over it, whoops

**Review by TiviPlus — CHANGES_REQUESTED** [review]
This is just copy paste defender sweep

**Comment by coldironwarrior** [comment]
> This is just copy paste defender sweep

yes which is why i moved the ability off defender's datum and into generic

**Comment by MiniMeatwad** [comment]
It already has a stun, use eggs. Forces a different playstyle and doesn't just make it click marine till go horizontal

**Comment by coldironwarrior** [comment]
just gonna close, too tired to sort out fixing up the code rn but if anyone wants to try this, i think ?????????? kurt would still want to see this/consider it?

**Pull request closed without merging**